src/app/problem/editor/right/ProvidedCodes.js

```diff
@@ -0,0 +1,98 @@
+import Block from "./Block";
+import {useEffect, useState} from "react";
+import {EditSaveCancelButton} from "../../commons/EditSaveCancelButton";
+import {useUploads} from "../../../usecases/UploadFilesUseCase";
+import {ACTION_UPDATE_PROVIDEDCODES, useProblemEditorContext} from "../context";
+import IconTextItems from "../../../commons/TextInputForm/IconTextItems";
+import FixedUploadFileItems from "./FixedUploadFileItems";
+import UploadFileButton from "../../commons/UploadFileButton";
+import {VscFileCode} from "react-icons/all";
+import {problemService} from "../../../../services/services";
+
+
+const ProvidedCodes = () => {
+    const {problem, dispatch} = useProblemEditorContext();
+    const [isEditing, setIsEditing] = useState(false);
+    const [saving, setSaving] = useState(false);
+    const [providedCodes, setProvidedCodes] = useState(undefined);
+    const {files, setFiles, addFiles, removeFile} = useUploads();
+
+    useEffect(() => {
+        if (problem) {
+            if (!providedCodes) {
+                /* ProvidedCodes may not exist in languageEnv,
+                 * If providedCodes doesn't exist, set an empty array for providedCodesFileNames
+                 */
+                if (problem.languageEnvs[0].providedCodes) {
+                    setProvidedCodes(problem.languageEnvs[0].providedCodes);
+                } else {
+                    setProvidedCodes({providedCodesFileNames: [], providedCodes: ""});
+                }
```

Comment:
Rewrite to
`setProvidedCodes(problem.languageEnvs[0].providedCodes || {fileId: "", fileNames: []});`

Note that I rename `providedCodesFileNames` to `fileNames` and I guess you mistype `fileId` to `providedCodes`

Reply:
Nice catch! Thanks for finding my typo. 😄

Reply:
Still unsolved, you should rename `providedCodesFileNames` to `fileNames`

but then I found that you did it wrong in backend.

Please first resolve this issue Judge-Girl/Judge-Girl#421